extensions/applicationinsights-osplugin-js/src/OsPlugin.ts

```diff
@@ -0,0 +1,253 @@
+
+        let _eventQueue: IDelayedEvent[];
+        let _evtNamespace: string | string[];
+        let _excludePageUnloadEvents: string[];
+
+        let _os: string;
+        let _osVer: number;
+        let _retryTime: number;
+    
+        dynamicProto(OsPlugin, this, (_self, _base) => {
+
+            _initDefaults();
+
+            _self.initialize = (coreConfig: IConfiguration & IConfig, core: IAppInsightsCore, extensions: IPlugin[]) => {
+                let _self = this;
+                _core = core;
+                super.initialize(coreConfig, core, extensions);
+                let identifier = _self.identifier;
+                _evtNamespace = mergeEvtNamespace(createUniqueNamespace(identifier), core.evtNamespace && core.evtNamespace());
+
+                _os = sessionStorage.getItem("ai_os");
+                _osVer = parseInt(sessionStorage.getItem("ai_osVer"));
+                if(_os && _osVer){
+                    _retrieveFullVersion = true;
+                }
+                _self._addHook(onConfigChange(coreConfig, (details)=> {
+                    let coreConfig = details.cfg;
+                    let ctx = createProcessTelemetryContext(null, coreConfig, core);
+                    _ocConfig = ctx.getExtCfg<IOSPluginConfiguration>(identifier, defaultOSConfig);
+                    _getOSTimeoutMs = _ocConfig.getOSTimeoutMs;
+                    _endpointIsBreeze = _ocConfig.endpointIsBreeze;
+                  
+                    let excludePageUnloadEvents = coreConfig.disablePageUnloadEvents || [];
+
+                    if (_excludePageUnloadEvents && _excludePageUnloadEvents !== excludePageUnloadEvents) {
+                        removePageUnloadEventListener(null, _evtNamespace);
+                        removePageHideEventListener(null, _evtNamespace);
+                        _excludePageUnloadEvents = null;
+                    }
+                   
+                    if (!_excludePageUnloadEvents) {
+                        // If page is closed release queue
+                        addPageUnloadEventListener(_doUnload, excludePageUnloadEvents, _evtNamespace);
+                        addPageHideEventListener(_doUnload, excludePageUnloadEvents, _evtNamespace);
+                    }
+                    _excludePageUnloadEvents = excludePageUnloadEvents;
+                }));
+                function _doUnload() {
+                    _releaseEventQueue();
+                }
+            };
+        
+            _self.processTelemetry = (event: ITelemetryItem, itemCtx?: IProcessTelemetryContext) => {
+                itemCtx = _self._getTelCtx(itemCtx);
+
+                if (!_retrieveFullVersion && !_getOSInProgress && _retryTime < maxRetry) {
+                    // Start Requesting OS version process
+                    _getOSInProgress = true;
+                    // Timeout request if it takes more than 5 seconds (by default)
+                    _getOSTimeout = setTimeout(() => {
```

Comment:
Lets move the creation of the timer into the `startRetrieveOsVersion` so it doesn't try to create a timer unless it's needed